src/tests/ftest/soak/soak.py

```diff
@@ -588,16 +620,22 @@ def create_fio_cmdline(self, job_spec, pool):
                     if fio_cmd.api.value == "POSIX":
                         # Connect to the pool, create container
                         # and then start dfuse
-                        self.start_dfuse(pool)
+                        dfuse, cmds = self.start_dfuse(pool)
                         fio_cmd.update(
                             "global", "directory",
-                            self.dfuse.mount_dir.value,
+                            dfuse.mount_dir.value,
                             "fio --name=global --directory")
-                    # fio command
                     log_name = "{}_{}_{}".format(blocksize, size, rw)
-                    commands.append([fio_cmd.__str__(), log_name])
-                    self.log.info(
-                        "<<FIO cmdline>>: %s \n", commands[-1])
+                    # self.dfuse_list.append(dfuse)
+                    cmds.append(slurm_utils.srun_str(
+                        hosts=None,
+                        cmd=str(fio_cmd.__str__()),
+                        srun_params=None))
+                    cmds.extend(self.stop_dfuse(dfuse))
+                    commands.append([cmds, log_name])
+                    self.log.info("<<Fio cmdlines>>:")
+                    for cmd in cmds:
+                        self.log.info("%s", cmd)
```

Comment:
Should this be unindented so that the entire `cmds` list is displayed once?

Reply:
The cmdlist for each FIO cmd does look correct as is;  It prints out each set of slurm cmds after the FIO params are updated for each FIO job; no changes necessary

Reply:
Ah, `cmds` is reset on line on 615, but only if using `POSIX`.  So `cmds` will be end up inclduing multiple loops of commands if not using `POSIX`.   Also don't you want to only include stop dfuse commands if using `POSIX`?  Maybe this would work:
```
                    srun_cmds = []
					
                    # start dfuse if api is POSIX
                    if fio_cmd.api.value == "POSIX":
                        # Connect to the pool, create container
                        # and then start dfuse
                        dfuse, srun_cmds = self.start_dfuse(pool)
                        fio_cmd.update(
                            "global", "directory",
                            dfuse.mount_dir.value,
                            "fio --name=global --directory")
					
                    srun_cmds.append(slurm_utils.srun_str(
                        hosts=None,
                        cmd=str(fio_cmd.__str__()),
                        srun_params=None))

                    if fio_cmd.api.value == "POSIX":
                        srun_cmds.extend(self.stop_dfuse(dfuse))
                    
                    log_name = "{}_{}_{}".format(blocksize, size, rw)
                    commands.append([srun_cmds, log_name])
					
                    self.log.info("<<Fio cmdlines>>:")
                    for cmd in srun_cmds:
                        self.log.info("%s", cmd)
```